### PR TITLE
Staging fixes6

### PR DESCRIPTION
**Pull request by Bod9001**
and made the spark light affect better
### Changelog:

CL: [Fix] Shuttle console now updates position properly
CL: [Fix] foam doesn't go into space matrix randomly

### CODE REVIEW

File: UnityProject/Assets/Scripts/Objects/Consoles/ShuttleConsole.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@ namespace Objects.Shuttles
 	public class ShuttleConsole : NetworkBehaviour, ICheckedInteractable<HandApply>, IServerSpawn
 	{
 		public MatrixMove ShuttleMatrixMove;
-		private RegisterTile registerTile;
+		public RegisterTile registerTile;
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Objects/Consoles/ShuttleConsole.cs]:
```suggestion
		[NonSerialized]
		public RegisterTile registerTile;
```

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Objects/Consoles/ShuttleConsole.cs]: or make it a property with private setter

**Comment by ThatDan123** [comment]
Can you add a test for the offset?

File: UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/ReactionManager.cs [hunk 2]
```diff
@@ -186,10 +186,17 @@ private void ProcessWindNodes(MetaDataNode windyNode)
...
+					pushable.NewtonianPush(registerTile.transform.rotation * (Vector2)windyNode.WindDirection, Random.Range((float)(correctedForce * 0.8), correctedForce),  spinFactor: Random.Range(1, 150));
+				}
+
+
 				if (pushable.stickyMovement && windyNode.WindForce > 3)
 				{
-					if (pushable is MovementSynchronisation && windyNode.WindForce < (int)WindStrength.STRONG) continue;
+					var Playermove = pushable as MovementSynchronisation;
+					if ( windyNode.WindForce < (int)WindStrength.STRONG && Playermove == null) continue;
+					if (Playermove.CanBeWindPushed == false) continue;
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/ReactionManager.cs]:
```suggestion
					var playerMove = pushable as MovementSynchronisation;
					if ( windyNode.WindForce < (int)WindStrength.STRONG && playerMove == null) continue;
					if (playerMove.CanBeWindPushed == false) continue;
```

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/ReactionManager.cs]: And the ```&& playerMove == null``` is that in the right place? or is it meant to be ```||```

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/ReactionManager.cs]: null check is more expensive than magnitude

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/ReactionManager.cs]: yeah, not asking about sequence. You are checking playerMove after that If and you are not returning if it is null before. Is that supposed to be an ```||```?

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/ReactionManager.cs]: well the as operation just casts, and then you have to check it to make sure it is not null

**Pull request merged**